Automattic/wp-calypso
# FSE: Homepage Posts block sync

#### Changes proposed in this Pull Request

- added a new script for syncing shared files from https://github.com/automattic/newspack-blocks
- always to be used one way — from newspack to FSE, newly added readmes should make it clear

There are two options to use it:
- with a release (downloads a build artefact newspack-blocks.zip)
- with a branch (downloads the whole repo code from branch specified)

Sync script usage should be covered by readme file

#### Building code

After running: `lerna run build:homepage-posts --scope='@automattic/full-site-editing' --stream`

…`homepage-posts/dist` directory is getting filled with build assets.

#### Testing instructions

- `cd app/full-site-editing`
- `npm run sync:homepage-posts` should print a usage
- `npm run sync:homepage-posts --branch=master` or `npm run sync:homepage-posts --release=1.0.0-alpha.17` should both populate `full-site-editing-plugin/homepage-posts` with a single block and some other supporting files
- `lerna run build:homepage-posts --scope='@automattic/full-site-editing' --stream` to build both `editor` and `view` assets
- observe `homepage-posts/dist` and confirm there are assets present

At this point you've confirmed the sync and build systems work. Let's test the functionality:

- rollback any possible changes from previous testing and run build again to have up-to-date assets
- open up the editor
- confirm it's present and insert the Homepage Posts block
- it should load some posts, give you a preview of them, and you should be able to tweak setting (for example switch between list & grid)
- preview that post/page
- on frontend, you should see the block working, showing you latest posts as configured
- try previewing the grid view — if you can see posts in grid in both the editor and the frontend, it confirms proper CSS assets were loaded

#### Reviewing

Keep in mind everything in `newspack-homepage-articles` is a code that lives someplace else and in this repository we cannot do adjustments to it. It has been already reviewed before being included in Newspack Blocks.
